test-d/router-test-d.ts

```diff
@@ -46,6 +46,9 @@ import { ERes } from '../dist/lib/routing/response';
   router.get('/', async (req, res) => {
     expectType<MyReq>(req)
     expectType<WoofRes>(res)
-    return res.send("Hello world!");
+    res.on('finish', finalResponse => {
+      expectType<Response | undefined>(finalResponse)
+    })
```

Comment:
This would be part of my suggestion to emit this event using `EResponse` instead of `Response`. Only do this if you are taking that suggestion.

```suggestion
    res.on('finish', e => {
      expectType<EResponse | undefined>(e)
    })
```